pennylane/drawer/tape_mpl.py

```diff
@@ -271,19 +262,66 @@ def _tape_mpl(tape, wire_order=None, show_all_wires=False, decimals=None, *, fig
         for op in layer_ops:
             _add_operation_to_drawer(op, drawer, layer, config)
 
-    for wire in _get_measured_wires(tape.measurements, list(range(n_wires))):
-        drawer.measure(n_layers, wire)
+    if measurements == "dots":
+        for wire in wire_map.values():
+            drawer.ax.text(
+                n_layers + 0.25,
+                wire,
+                s="...",
+                ha="center",
+                va="center_baseline",
+                fontsize=1.5 * drawer.fontsize,
+            )
+    else:
+        for wire in _get_measured_wires(measurements, list(range(n_wires))):
+            drawer.measure(n_layers, wire)
 
-    measured_bits = _get_measured_bits(tape.measurements, bit_map, drawer.n_wires)
-    if measured_bits:
-        drawer.measure(n_layers, measured_bits)
+        measured_bits = _get_measured_bits(measurements, bit_map, drawer.n_wires)
+        if measured_bits:
+            drawer.measure(n_layers, measured_bits)
 
     return drawer.fig, drawer.ax
 
 
+def _tape_mpl(tape, wire_order=None, show_all_wires=False, max_length=None, **kwargs):
+    """Private function wrapped with styling."""
+    wire_map = convert_wire_order(tape, wire_order=wire_order, show_all_wires=show_all_wires)
+    tape = transform_deferred_measurements_tape(tape)
+    tape = qml.map_wires(tape, wire_map=wire_map)[0][0]
+    bit_map = default_bit_map(tape)
+
+    layers = drawable_layers(tape.operations, wire_map={i: i for i in tape.wires}, bit_map=bit_map)
+
+    for i, layer in enumerate(layers):
+        if any(isinstance(o, qml.measurements.MidMeasureMP) and o.reset for o in layer):
+            layers.insert(i + 1, [])
+
+    if max_length is None:
+        return _draw_layers(layers, tape.measurements, bit_map=bit_map, wire_map=wire_map, **kwargs)
+
+    figs_and_axes = []
+    while layers:
+        next_slice = layers[:max_length]
+        layers = layers[max_length:]
+        measurements = "dots" if layers else tape.measurements
+        figs_and_axes.append(
+            _draw_layers(next_slice, measurements, bit_map=bit_map, wire_map=wire_map, **kwargs)
+        )
```

Comment:
```suggestion
    layer_count = len(layers)
    figs_and_axes = [
        _draw_layers(
            layers=layers[i : i + max_length],
            measurements=tape.measurements if i + max_length >= layer_count else "dots",
            bit_map=bit_map,
            wire_map=wire_map,
            **kwargs,
        )
        for i in range(0, layer_count, max_length)
    ]
```
This might be better to avoid repeatedly updating `layers`, and also more straightforward which slice we are refering to in the original layers list. if it's fine to keep `layers` as it is.

Reply:
adding locally.